### PR TITLE
Renaming services for Kafka, Kafka Connect and Zookeeper

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- ~~Bugfix~~
- Enhancement / new feature
- ~~Refactoring~~

### Description

This PR renames the services as discussed in #55.

`xxx-kafka` -> `xxx-kafka-bootstrap`
`xxx-kafka-headless` -> `xxx-kafka-brokers`
`xxx-zookeeper` -> `xxx-zookeeper-client`
`xxx-zookeeper-headless` -> `xxx-zookeeper-nodes`
`xxx-connect` -> `xxx-connect-api`

It also adds new `CHANGELOG.md` file with the list of latest updates and important changes.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java [hunk 1]
```diff
@@ -87,7 +88,7 @@
     private static final boolean DEFAULT_KAFKA_METRICS_ENABLED = false;
 
     // Kafka configuration defaults
-    private static final String DEFAULT_KAFKA_ZOOKEEPER_CONNECT = "zookeeper:2181";
+    private static final String DEFAULT_KAFKA_ZOOKEEPER_CONNECT = "zookeeper-client:2181";
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java]: Taking a look at the code I was just wondering if this default makes sense to have. If Zookeeper is deployed using Strimzi, it's name will be always something line `xxx-zookeeper-client` and you now have changed the assignment as follow : 

`result.setZookeeperConnect(ZookeeperCluster.serviceName(kafkaAssembly.getMetadata().getName()) + ":2181");`

The default is really used just for this declaration : 

`private String zookeeperConnect = DEFAULT_KAFKA_ZOOKEEPER_CONNECT;`

which is overridden by above. Maybe we don't need it.

Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java]: Yeah, you are right ... in reality it is never really used. We always change the value.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# CHANGELOG
```
Comment (tombentley) [inline CHANGELOG.md]: We should add 'Update CHANGELOG' to be PR checklist.

Comment (scholzj) [reply CHANGELOG.md]: Yeah, makes sense. Good point.

**Pull request merged**